pkg/lib/esbuild-checksum-plugin.js

```diff
@@ -0,0 +1,45 @@
+import fs from "node:fs/promises";
+import path from 'node:path';
+
+import sha256 from "js-sha256";
+
+const NAME = 'cockpitChecksumPlugin';
+
+const getAllFiles = async function(dirPath, arrayOfFiles) {
+    for (const file of await fs.readdir(dirPath)) {
```

Comment:
I don't like `fs.readdir()` here.  Is there really no way to get the list of files from esbuild?

Reply:
You can run esbuild in `write: false` mode, then it doesn't write any artifacts and instead `result.outputFiles` has a map of file name → binary contents. I was playing around with that yesterday, as that would be a much nicer approach for this case. However, using this requires re-engineering all our other plugins. I would like to do this some day, but it's quite a lot of work.

Reply:
... Thus for now it uses the same approach as https://github.com/cockpit-project/cockpit/blob/main/pkg/lib/esbuild-compress-plugin.js

Reply:
Can you open an issue to port the po-plugin and drop our custom compression plugin and change this one to operate on the map?

Reply:
This cannot be determined by-plugin -- it's a global setting. *all* plugins need to move over from "file" to "in-memory map" processing at the same time. I can look at that this afternoon, but it's not super-easy -- for example, compression would also happen in JS then instead of calling `gzip`.

Reply:
Filed issue #19039 for that.